src/azure/storage/signer.rs

```diff
@@ -247,11 +279,19 @@ pub enum SignedOutput {
 /// CanonicalizedHeaders +
 /// CanonicalizedResource;
 /// ```
+/// ## Note
+/// For sub-requests of batch API, requests should be signed without `x-ms-version` header.
+/// Set the `sub_req` to `ture` for such.
 ///
 /// ## Reference
 ///
 /// - [Blob, Queue, and File Services (Shared Key authorization)](https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key)
-fn string_to_sign(req: &impl SignableRequest, cred: &Credential, now: DateTime) -> Result<String> {
+fn string_to_sign(
+    req: &impl SignableRequest,
+    cred: &Credential,
+    now: DateTime,
+    sub_req: bool,
```

Comment:
**NOTICE**: The signer is not familiar with the term "sub request". Please use a more specific name that is relevant to the current context, such as `omit_service_version`. It's important to remember that not all users or projects may have the same understanding of terminology as you do.